Cross-platform

jpenilla opened the pull request:
Moves most API to a shared module between adventure-platform-fabric and adventure-platform-neoforge.

Todos:
- [x] Clean up build scripts
- [ ] Full audit of diff & comparison of platforms, testing all behavior on both platforms
  - [x] server-side localizations
  - Audience 
    - [ ] clearResourcePacks
    - [ ] clearTitle
    - [ ] deleteMessage
    - [x] hideBossBar
    - [x] openBook
    - [x] playSound (emitters)
    - [ ] removeResourcePacks
    - [ ] resetTitle
    - [x] sendActionBar
    - [x] sendMessage
    - [x] sendPlayerListHeader,Footer,HeaderAndFooter
    - [ ] sendTitlePart
    - [x] showBossBar
    - [ ] showTitle
    - [ ] stopSound
- [x] Testing of both platforms published components in external projects (in-dev and prod)
- [x] Check for anything else that can be moved from mixins to API on neo
- [x] Review existing APIs to see if there's anything else we want to break before 6.0 (off the top of my head stuff like MinecraftAudiences.update is kind of broken due to being static and lacking registry/audience context, and mc Component being ComponentLike may want to be revisited for the same reason)

... and probably more that I'm forgetting

jpenilla committed:
Move fabric into subproject

jpenilla committed:
Initial neoforge & common project setup

jpenilla committed:
More build setup

jpenilla committed:
move lots of server code to common

jpenilla committed:
wip

built jar loads on neoforge client (testmod broken in dev currently)

jpenilla committed:
more wip

jpenilla committed:
more wip

jpenilla committed:
more wip

jpenilla committed:
switch from archloom to moddevgradle

jpenilla committed:
hack to make it work in-dev

jpenilla committed:
license and style fixes

now builds except for broken tests

jpenilla committed:
switch from archloom to fabric loom

jpenilla committed:
fix unit tests

jpenilla committed:
Move locale tracking injection to fabric in anticipation of neoforge adding api

jpenilla committed:
Initialize default locale

jpenilla committed:
setup fabric repack

jpenilla committed:
Bump version to 6.0.0-SNAPSHOT

jpenilla committed:
Add more casting helpers and un-deprecate existing helpers/conversion methods

jpenilla committed:
Remove remaining fabric references in common module and add javadoc for new methods

jpenilla committed:
Move more API to common

colonel impl is fabric-only for now

jpenilla committed:
remove redundant overrides

jpenilla committed:
update build.yml

jpenilla committed:
update renovate.json

jpenilla committed:
switch to upstream loom

jpenilla committed:
disable mixin ap on fabric repack

jpenilla committed:
apply parchment uniformly

zml2008 commented:
> Should all since tags in relocated/renamed classes be updated?

imo yes, since the changes are not api-compatible

jpenilla committed:
Update since tags

jpenilla committed:
reapply idea ext plugin and move some deps to version catalog

jpenilla committed:
move HiddenRequirement injection to mod-shared

jpenilla committed:
setup buildSrc and move repeated dependency declarations

jpenilla committed:
apply checkstyle in convention plugin

jpenilla committed:
remove some redundant plugin declarations

jpenilla committed:
extract common publishing/javadoc logic

jpenilla committed:
fix mod-shared (not repacked) not publishing

jpenilla committed:
Make MinecraftAudiences.update non-static

fixes issues with registry-aware components

jpenilla committed:
n.m.n.c.Component is no longer ComponentLike

the lack of context means this won't always work (it will fail at runtime depending on the data passed by other mods/datapacks/etc)

jpenilla committed:
fix fabric tester compile

jpenilla committed:
fix find-replace errors

jpenilla committed:
neoforge: listen to ClientInformationUpdatedEvent

jpenilla committed:
build: fix root project check

jpenilla committed:
Remove PlayerChatMessage interface injection and make conversion method non-static to inject context

jpenilla committed:
use entity's registry access for hover event conversion

jpenilla committed:
Use player's registry access for display name conversion

jpenilla committed:
Update more since tags

jpenilla committed:
Make MinecraftAudiences#nonWrappingSerializer not static and return a registry-aware converter

jpenilla committed:
Apply our renderer if one isn't attached in MinecraftAudiences#update

jpenilla committed:
cleanup diff from previous commit

jpenilla committed:
expand neoforge tester, rename some methods for consistency, various fixes

jpenilla committed:
Deduplicate test resources

jpenilla committed:
Fix HiddenRequirement not being hidden on NeoForge

jpenilla committed:
checkstyle

jpenilla committed:
Move API mixins to -fabric exclusive, bring back service loader workaround

Fixes non-game layer libs being unable to access adventure classes on NeoForge

jpenilla committed:
`toNative` -> `asNative`

jpenilla committed:
Make more MinecraftAudiences methods PolyNull

jpenilla committed:
rename fabric tester for clarity

jpenilla committed:
Fix argument type serialization on neo

jpenilla committed:
Remove deprecated APIs

jpenilla committed:
brig/authlib mixins don't work on neo

jpenilla committed:
Implement PermissionChecker pointer on NeoForge

jpenilla committed:
Implement MinecraftServerAudiencesImpl#permission(String)

jpenilla committed:
Add helper for registering adventure argument types on NeoForge

jpenilla committed:
fix checkstyle

jpenilla committed:
Bump NeoForge

jpenilla committed:
Actually set new locale on NeoForge

jpenilla committed:
Update ModDevGradle

jpenilla committed:
Update README

zml2008 committed:
misc tweaks?

zml2008 commented:
The repack variant, as mentioned in the docs

> A second variant, net.kyori:adventure-platform-mod-shared-fabric-repack, is also published. This variant should be used when your common code is managed by fabric-loom.

does not seem to be published:

```diff
diff --git a/buildSrc/src/main/kotlin/publishing-conventions.gradle.kts b/buildSrc/src/main/kotlin/publishing-conventions.gradle.kts
index 9d2ff18..49fa985 100644
--- a/buildSrc/src/main/kotlin/publishing-conventions.gradle.kts
+++ b/buildSrc/src/main/kotlin/publishing-conventions.gradle.kts
@@ -11,6 +11,15 @@ indraCrossdoc {
   }
 }
 
+publishing {
+  repositories {
+    maven {
+      url = rootProject.projectDir.resolve("test-repo").toURI()
+      name = "projectLocal"
+    }
+  }
+}
+
 tasks {
   javadoc {
     val options = this.options as StandardJavadocDocletOptions
```

then `$ ./gradlew pAPTPLMR`

check `$rootDir/test-repo` to see what's published

zml2008 commented on neoforge/src/main/java/net/kyori/adventure/platform/neoforge/impl/AdventureNeoforgeCommon.java:
```diff
@@ -0,0 +1,170 @@
+
+    @Override
+    public void collectPointers(final Pointered pointered, final Pointers.Builder builder) {
+      PlatformHooks.super.collectPointers(pointered, builder);
```
Does it make sense to call a neo-specific event here?

zml2008 commented on neoforge/gradle.properties:
```diff
@@ -0,0 +1 @@
+loom.platform=neoforge
```
Still necessary with MDG?

zml2008 commented on neoforge/build.gradle.kts:
```diff
@@ -0,0 +1,81 @@
+  processResources {
+    val props = mapOf("version" to project.version)
+    inputs.properties(props)
+    filesMatching("META-INF/neoforge.mods.toml") {
```
Maybe use blossom instead? Long-term goal with KyoriPowered/blossom#39 is to move the yaml -> json fuckery we do on the fabric side into blossom (or a companion plugin) as well.

zml2008 commented on build.gradle:
```diff
@@ -0,0 +1,102 @@
+  }
+
+  indra {
+    github("KyoriPowered", "adventure-platform-fabric") {
```
heads up for us all, once this PR is merged we should remember to rename the repo and its references to adventure-platform-mod.

zml2008 commented on build.gradle:
```diff
@@ -0,0 +1,102 @@
+
+          developer {
+            id = "Electroid"
+          }
```
@jpenilla please add your information here

zml2008 commented on build.gradle:
```diff
@@ -0,0 +1,102 @@
+import net.fabricmc.loom.task.RunGameTask
```
out of curiosity, why put this configuration here instead of in a buildSrc convention plugin?

jpenilla replied:
Mainly laziness since loom was already on the classpath from the settings (and isn't in buildSrc), also I think it was like this in your draft branch that I started with.

Since we use `repositoriesMode.set(RepositoriesMode.PREFER_SETTINGS)` and therefore need to apply loom to the settings regardless, we would need to declare the loom version in the version catalog and settings to move it. (Unless we use some complicated work around like https://discuss.gradle.org/t/how-to-use-version-catalog-in-the-root-settings-gradle-kts-file/44603/5)

jpenilla replied:
~~Actually, since we don't use loom for NeoForge anymore, it may be simpler to resolve. Since we don't care about runs in the repack project.~~ The MC dependency declarations are also in the root build, not just the run configurations. But it's just a couple of lines so I'll dupe it to clean up the root script.

jpenilla committed:
Disable javadoc generation/publishing for mod-shared-fabric-repack

jpenilla committed:
Add my info

jpenilla committed:
Remove no longer needed file

this was for archloom, we are using MDG now

jpenilla commented:
The repack project is published, but it doesn't use the publishing conventions as I didn't want to configure crossdoc (or really even enable the Javadoc jar) for it. Even if I set up Javadoc to build off of the remapped source, a duplicate of the -shared JD using intermediary names doesn't seem useful.

jpenilla committed:
Move loom configuration out of root project

jpenilla committed:
Don't apply configurateTransformations plugin to :neoforge

jpenilla committed:
NeoForge: Add CollectPointersEvent

zml2008 committed:
fix: Avoid publishing empty JD for neoforge-services

zml2008 committed:
chore(fabric): Disable publication until we target 1.21.2
